### DISCUSSION TITLE
How to cancel workflow instance with loop? What means "canceleing" worflow instance?

### DISCUSSION THREAD

**Discussion by gor8808** [elsa-workflows/elsa-core#3053] [Q&A] [answered]
I have this workflow 

While(true) 
  |(iterate)|
Writeline()

When I cancel this workflow (after execution) using this endpoint -- ​/v1​/workflow-instances​/{id}​/cancel it still keeps executing(printing to console)

Can you specify what means canceling the instance and in which cases it cancels the instance? 
BTW I can't find any docs regarding canceling endpoint

**Comment by sfmskywalker** [comment, marked as answer]
Hi @gor8808 

Canceling a workflow does not currently cancel "hot" workflow instances (where an activity keeps generating work).
Right now, cancelling a workflow will only prevent it from being resumed.
But I can see that it would be useful to be able to cancel a workflow even while it executes.

I'll create an issue for this.

**Reply by gor8808** [reply]
Hi @sfmskywalker, okay thanks for answering
